rpc/src/v1/impls/eth.rs

```diff
@@ -667,7 +687,9 @@ impl<C, SN: ?Sized, S: ?Sized, M, EM, T: StateInfo + 'static> Eth for EthClient<
 		};
 
 		let transaction_id = PendingTransactionId::Location(block_id, index.value());
-		Box::new(future::done(self.transaction(transaction_id)))
+		let result = self.transaction(transaction_id)
```

Comment:
Here we actually have `BlockNumber` so we don't have to fallback to this heuristic for hashes (just check if the block number is available and return error if not).